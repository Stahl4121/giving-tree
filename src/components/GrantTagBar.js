import React from 'react';
import { Link } from 'react-router-dom';
import { Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  button: {
    borderRadius: '25px',
    margin: '2px',
  },
}))

class GrantTagBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tags: props.tags,
    };
  };


  render() {

    //const tags = [];

    const tagElements = []

    for (const [index, value] of this.state.tags.entries()) {
      tagElements.push(<GrantTag tag={value} />)
    }

    return (
      tagElements
    );
  }
}



function GrantTag(props) {
  const classes = useStyles();

  return (
    <Link
      to={{
        pathname: "/",
        state: { incomingTag: props.tag }
      }}
    >
      <Button className={classes.button} size='small'  variant='outlined'>
        {props.tag}
      </Button>
    </Link>
  );
}

export default GrantTagBar;