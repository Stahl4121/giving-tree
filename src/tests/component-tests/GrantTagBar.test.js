import React from 'react';
import { shallow, mount } from 'enzyme';

import GrantTagBar from '../../components/GrantTagBar.js';
import { StaticRouter } from 'react-router-dom';
import GrantTag from '../../components/GrantTagBar.js';


describe('Component: GrantTagBar', () => {

  it('State: no tags', () => {
    const gtbProps = {
      tags: []
    };
    const context = {};
    const wrap = mount(
      <StaticRouter location="someLocation" context={context}>
        <GrantTagBar id={45} tags={gtbProps.tags} />
      </StaticRouter>
    )
    //const tree = wrap.toJSON();
    expect(wrap.find(GrantTagBar).state('tags')).toEqual([])
  })


  it('State: one tag', () => {
    const gtbProps = {
      tags: ['Tag']
    };
    const context = {};
    const wrap = mount(
      <StaticRouter location="someLocation" context={context}>
        <GrantTagBar tags={gtbProps.tags} />
      </StaticRouter>
    )
  
    expect(wrap.find(GrantTagBar).state('tags')).toEqual(['Tag'])
  })

  it('State: two tags', () => {
    const gtbProps = {
      tags: ['First', 'Second']
    };
    const context = {};
    const wrap = mount(
      <StaticRouter location="someLocation" context={context}>
        <GrantTagBar id={45} tags={gtbProps.tags} />
      </StaticRouter>
    )
  
    expect(wrap.find(GrantTagBar).state('tags')).toEqual(['First', 'Second'])
  })
  
  
})


describe('Component: GrantTag', () => {

  it('Tag Text', () => {
    const tag = 'Tag'
    const context = {};
    const wrap = shallow(
      <StaticRouter location="someLocation" context={context}>
        <GrantTag tag={tag} />
      </StaticRouter>
    )
    expect(wrap.find(GrantTag).prop('tag')).toEqual('Tag')
  })


  /*it('Link takes to Dashboard', () => {
    const tag = 'Tag'
    const context = {};
    const wrap = shallow(
      <StaticRouter location="someLocation" context={context}>
        <GrantTag tag={tag} />
      </StaticRouter>
    )
    expect(wrap.find(GrantTag).find('Link').prop('to')).to.be.equal('/mission')
  })

  it('Link sends tag to Dashboard', () => {
    const tag = 'Tag'
    const context = {};
    const wrap = shallow(
      <StaticRouter location="someLocation" context={context}>
        <GrantTag tag={tag} />
      </StaticRouter>
    )
    expect(wrap.state('Tag')).toEqual('Tag')
  })*/
  
  
})