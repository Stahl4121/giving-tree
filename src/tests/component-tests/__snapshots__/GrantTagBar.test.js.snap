// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: GrantTag Renders correctly 1`] = `ShallowWrapper {}`;

exports[`Component: GrantTagBar Renders correctly 1`] = `ShallowWrapper {}`;
